### ISSUE TITLE
My Nginx Proxy Manager does not renew SSL certificates after expiry

### ISSUE THREAD

**Issue by uhsa779** [jc21/nginx-proxy-manager#1447]
I am assuming I am not the only 1 having this issue. I tried to see if anyone had any resolution but didn't seem so. 

The only reason I am opening my own "issue" is I didnt want to hijack someone else's report. I apologize if this isn't the correct way. 

A bit about my issue:

- Docker/Portainer NPM install (https://github.com/jlesage/docker-nginx-proxy-manager)
- Certbot 1.19.0
- My SSL certs expired - and I tried to renew it - keeps failing on each.
- I deleted some certs hoping I could create new ones - fails.
- I deleted the proxy host, deleted certs and tried to re-create new ones - failed.

I am happy to include anything you guys would like me to - sort of a noob here so may take a bit of pointing me to the right direction...
![letsencrypt](https://user-images.githubusercontent.com/79544342/135636854-dbd03185-b555-4180-b51c-0ad31f4f91ff.JPG)

**Comment by cybergurke** [comment]
You are really bad at censoring your domain.

**Comment by uhsa779** [comment]
> You are really bad at censoring your domain.

Not sure if you are being sarcastic or if this somehow helps me in fixing my issue...

**Comment by cybergurke** [comment]
Sorry I cant help you. Im here because of the same Issue. It looks like you want to hide your Domain, but its clearly visible at the second line.

**Comment by uhsa779** [comment]
> Sorry I cant help you. Im here because of the same Issue. It looks like you want to hide your Domain, but its clearly visible at the second line.

ah - that's ok. i am not fighting russian hackers. Just didn't want to have everything out in open - thanks for pointing out though. appreciate your concern.

**Comment by xannor** [comment]
There has been a recent issue with the Lets Encrypt SSL Root Cert expiring. Have you tried to purge and re-create the ssl certs? It could be they are not transitioning properly and thus failing on renewal.

**Comment by rainman369** [comment]
I'm getting similar errors when trying to issue a new cert on a google domain. Does it want us to use the "dns challenge" option when issuing the new cert?? If so, I have no idea how to fill it in.
![Screen Shot 2021-10-02 at 3 32 54 PM](https://user-images.githubusercontent.com/61566782/135729743-f32f37b5-e913-4cf9-b6b4-29c11627eacf.png)
![Screen Shot 2021-10-02 at 3 33 34 PM](https://user-images.githubusercontent.com/61566782/135729744-dd7cfb9a-44a0-45e3-876d-ac1eb2715d00.png)

**Comment by uhsa779** [comment]
Yes - I tried deleting them but no effect. I even deleted host and recreated them- nothing.  ---- On Sat, 02 Oct 2021 10:26:54 -0500  ***@***.***  wrote ----
There has been a recent issue with the Lets Encrypt SSL Root Cert expiring. Have you tried to purge and re-create the ssl certs? It could be they are not transitioning properly and thus failing on renewal.

**Comment by xannor** [comment]
@rainman369 I believe it just passes the field through to certbot so you would need to follow the directions for certbot... For google they can be found here... https://certbot-dns-google.readthedocs.io/en/stable/

**Comment by uhsa779** [comment]
I have pihole with unbound  - would that matter? ---- On Sat, 02 Oct 2021 15:28:17 -0500  ***@***.***  wrote ----
@rainman369 I believe it just passes the field through to certbot so you would need to follow the directions for certbot... For google they can be found here... https://certbot-dns-google.readthedocs.io/en/stable/

**Comment by xannor** [comment]
I dont know, I am not overly familiar with unbound. 
The import thing would be what /var/log/letsencrypt/letsencrypt.log says about the failure when you attempt to get a new cert, since what you have posted so far pertains to a renewal, and only states that it failed. Look through that, and if you cannot see an answer post it (scrubbed of course so you do leak any identifiers or passwords.) and maybe we can find you an answer.

**Comment by uhsa779** [comment]
> I dont know, I am not overly familiar with unbound. The import thing would be what /var/log/letsencrypt/letsencrypt.log says about the failure when you attempt to get a new cert, since what you have posted so far pertains to a renewal, and only states that it failed. Look through that, and if you cannot see an answer post it (scrubbed of course so you do leak any identifiers or passwords.) and maybe we can find you an answer.

Here's the output of the /var/log/letsencrypt/letsencrypt.log - I have removed all my domain mentions - all I could see was "firewall" which doesn't seem right. Also, all the domains in question are def registered and i am able to ping them tracing back to my IP. I tried disabling pihole/unbound (which basically takes me directly to public DNS) - but that doesn't help at all. 

I deleted and tried to re-create; no help there either. Nothing seems to work...

Thank you for helping out. 
[letsencrypt-var.log](https://github.com/jc21/nginx-proxy-manager/files/7280629/letsencrypt-var.log)
-

**Comment by uhsa779** [comment]
> There has been a recent issue with the Lets Encrypt SSL Root Cert expiring. Have you tried to purge and re-create the ssl certs? It could be they are not transitioning properly and thus failing on renewal.

Tried deleting proxy hosts as well as their SSL certs - still doesn't work. 
> 
> edit: I missed where you said you have removed the hosts. do you get a different error when you attempt to create new certs? Also have you checked the lets encrypt log file? it should be under your data volume.
Attached the log files above - both time it says "Internal Error" 
> 
> I also noticed you are using the jlesage image and not the "official" one, so I dont know if there is any difference in the lets encrypt install he provides.

That will be mylast option - delete the docker container and recreate it with "official" one...

**Comment by rainman369** [comment]
> I'm getting similar errors when trying to issue a new cert on a google domain. Does it want us to use the "dns challenge" option when issuing the new cert?? If so, I have no idea how to fill it in.
> ![Screen Shot 2021-10-02 at 3 32 54 PM](https://user-images.githubusercontent.com/61566782/135729743-f32f37b5-e913-4cf9-b6b4-29c11627eacf.png)
> ![Screen Shot 2021-10-02 at 3 33 34 PM](https://user-images.githubusercontent.com/61566782/135729744-dd7cfb9a-44a0-45e3-876d-ac1eb2715d00.png)

I was able to solve my quoted issue by selecting HTTPS instead of HTTP in nginx when applying for the certificate.

**Comment by uhsa779** [comment]
I did not have any luck when I change my schema from HTTP to HTTPS. 

I upgraded my container from the "official" "jc21" repo - still same issue. This time around I have to create brand new proxy hosts since I created a new container - and with this new one, it has it's own mariadb- so nothing was transferred over. 

Here's the latest attempt to have a brand new cert
[letsencrypt-jc21NPM.log](https://github.com/jc21/nginx-proxy-manager/files/7281142/letsencrypt-jc21NPM.log)
.
